### PR TITLE
chore: cleanup middleware implementation

### LINKED ISSUE #6968: Clean up middleware implementation
Middlewares work, but the code and tests are a bit confusing. Let's fix that.

### PR DESCRIPTION
**Pull request by williamluke4**
closes https://github.com/prisma/prisma/issues/6968

### CODE REVIEW

**Comment by millsp** [comment]
I think that we should keep the commits sequential (not re-write over one's commit). This is important so that we can preserve code history, and track code ownership. In this case, we can't see my changes as they have been overwritten:
```
chore(client, middleware): extract / clean impl
```

@williamluke4

**Comment by millsp** [comment]
The tests that are failing with missing `clientMethod` are probably related to the changes I've made in `_request` (`getPrismaClient.ts`). Looking into it ASAP.

**Comment by millsp** [comment]
So it turns out that we were leaking `clientMethod`. As it is not part of `QueryMiddlewareParams`, it is not something that is set to be shared to the user. When I refactored `_request`, I ensured that no excess properties are leaked to the user. Please let me know if it is not the intended behavior.

**Comment by janpio** [comment]
We should probably split the refactor, the formatting changes and the test cleanup into separate PRs.

File: src/packages/cli/src/__tests__/__snapshots__/studio.test.ts.snap [hunk 1]
```diff
@@ -36,7 +36,6 @@ Object {
             string: true,
           },
         },
-        clientMethod: with_all_field_types.create,
```
Comment (Jolg42) [inline src/packages/cli/src/__tests__/__snapshots__/studio.test.ts.snap:39]: Not sure why this is gone? 🤔

Comment (williamluke4) [reply src/packages/cli/src/__tests__/__snapshots__/studio.test.ts.snap:39]: Don't think it was used anywhere. [Related](https://github.com/prisma/prisma/pull/6969#issuecomment-848052935)

File: src/packages/client/src/generation/TSClient/helpers.ts [hunk 2]
```diff
@@ -87,6 +87,8 @@ export function getArgFieldJSDoc(
     const comment = JSDocs[action]?.fields[fieldName](singular, plural)
     return comment as string
   }
+
+  return undefined
```
Comment (Jolg42) [inline src/packages/client/src/generation/TSClient/helpers.ts:91]: Curious to know why this was needed?

**Comment by Jolg42** [comment]
I reverted some unrelated changes (formatting) to make this PR cleaner

File: src/packages/client/src/generation/TSClient/helpers.ts (hunk 2 above)
Comment (Jolg42) [reply src/packages/client/src/generation/TSClient/helpers.ts:91]: Maybe because I see ` "noImplicitReturns": true,` added to tsconfig

**Review by Jolg42 — APPROVED** [review]
Well done @williamluke4 & @millsp on this cleanup! 🎊

File: src/packages/client/src/generation/TSClient/helpers.ts (hunk 2 above)
Comment (williamluke4) [reply src/packages/client/src/generation/TSClient/helpers.ts:91]: @millsp Personal preference

**Pull request merged**